tests/rptest/tests/cluster_linking_e2e_test.py

```diff
@@ -117,3 +123,141 @@ def test_basic_ops(self):
             assert services.secondary.is_kafka, \
                 f"Expected Kafka service, got {services.secondary}"
             self.basic_ops(services)
+
+
+class ShadowLinkBasicTests(RedpandaTest):
+    def __init__(self, test_context, *args, **kwargs):
+        super().__init__(test_context=test_context,
+                         num_brokers=3,
+                         log_config=LoggingConfig(
+                             'info', logger_levels={'cluster_link': 'trace'}),
+                         *args,
+                         **kwargs)
+
+        self.test_context = test_context
+        self.admin_v2: AdminV2 = None
+        self.services: MultiClusterServices = None
+
+    def setUp(self):
+        self.services = MultiClusterServices(
+            self.test_context,
+            self.logger,
+            self.redpanda,
+            secondary_type=ServiceType.REDPANDA,
+            num_brokers=3)
+        self.services.setUp()
+        self.services.primary.service.enable_development_feature_support()
+        self.services.primary.service.set_cluster_config(
+            values={"development_enable_cluster_link": True})
+        self.admin_v2 = AdminV2(self.services.primary.service)
+
+    def list_all_shadow_links(
+        self, client: shadow_link_pb2_connect.ShadowLinkServiceClient
+    ) -> list[shadow_link_pb2.ShadowLink]:
+        resp = client.call_list_shadow_links(
+            req=shadow_link_pb2.ListShadowLinksRequest())
+        if resp.error() is not None:
+            raise resp.error()
+        return [link for link in resp.message().shadow_links]
+
+    def get_shadow_link(
+            self, client: shadow_link_pb2_connect.ShadowLinkServiceClient,
+            name: str) -> shadow_link_pb2.ShadowLink:
+        resp = client.call_get_shadow_link(
+            req=shadow_link_pb2.GetShadowLinkRequest(name=name))
+        if resp.error() is not None:
+            raise resp.error()
+        return resp.message().shadow_link
+
+    @cluster(num_nodes=6)
+    def test_service_exists(self):
```

Comment:
hopefully overtime we can remove this and it's implicit because we test the behavior of each endpoint and not the existance of them?